contrib/src/main/java/gov/nasa/jpl/aerie/contrib/streamline/core/Resources.java

```diff
@@ -38,22 +41,26 @@ private Resources() {}
    *   This method is idempotent; calling it multiple times is the same as calling it once.
    * </p>
    */
-  public static void init() {
-    currentTime();
+  public static void init(Instant planStart) {
+    CLOCK = resource(clock(ZERO));
+    ABSOLUTE_CLOCK = name(addToInstant(planStart, CLOCK), "Global Absolute Simulation Clock");
   }
 
   // TODO if Aerie provides either a `getElapsedTime` method or dynamic allocation of Cells, we can avoid this mutable static variable
-  private static Resource<Clock> CLOCK = resource(clock(ZERO));
+  private static Resource<Clock> CLOCK;
+  private static Resource<InstantClock> ABSOLUTE_CLOCK;
   public static Duration currentTime() {
-    try {
-      return currentValue(CLOCK);
-    } catch (Scoped.EmptyDynamicCellException | IllegalArgumentException e) {
```

Comment:
Nice to see this exception handler go away. Not entirely sure I follow what the paradigm shift was that made it obsolete

Reply:
In hindsight, it was obvious. I just had to put a call to `Resources.init()` (or build a streamline registrar) in the constructor for my unit tests, so the unit tests would re-init the system at the correct time. The unit test is just a model, so it's exactly the same paradigm that we use in production. I don't know why I didn't realize I could do that before. I think maybe I was originally trying to avoid the need for an `init` method altogether?